### ISSUE TITLE
errors/index.ts: generateErrorMessage helper incorrectly returns 'undefined'

### ISSUE THREAD

**Issue by davemo** [apollographql/apollo-client#9588]
I recently shipped an upgrade of `@apollo/client 3.5.10` to production and we started noticing errors in Sentry with `Error: undefined` which we hadn't seen previously. I did some tracing and think I've isolated the problem to [src/errors/index.ts](https://github.com/apollographql/apollo-client/blob/main/src/errors/index.ts#L23-L27).

It appears that the ternary condition for determining if an error already has a message is wrong.

**Intended outcome:**
- We're expecting to see an error with an actual message, if it exists, or `Error message not found.` as the ternary would suggest. 

**Actual outcome:**
- What we see is `Error: undefined`.

![Screen Shot 2022-04-07 at 2 13 48 PM](https://user-images.githubusercontent.com/69559/162271115-46afae41-1cd9-40f2-a2a2-522687ba021b.png)

```typescript
// our error logging function
export const logError = (err: ApolloErrorWithCtx) => {
  if (shouldLog()) {
    console.error(err, err?.gql?.loc?.source?.body)
  } else {
    const firstError = (err as any)?.graphQLErrors?.[0]
    const isForHumans = firstError?.extensions?.for_humans
    Sentry.captureException(err?.message ? err.message : err, {
      extra: { gql: err?.gql?.loc?.source?.body },
      level: isForHumans ? Severity.Warning : Severity.Error,
    })
  }
}
```

![Screen Shot 2022-04-07 at 2 22 46 PM](https://user-images.githubusercontent.com/69559/162271137-24476d09-c868-4974-a108-61ff911d95db.png)

**How to reproduce the issue:**
1. Setup a query and use `.subscribeToMore` with an `onError` handler that logs out the error (ie: `onError: (err) => { console.log(err); }`)
2. Reload your browser page that has apollo-client along with subscriptions configured.
3. Trigger a momentary offline state for your backend (we do this by saving a `.go` file, which triggers recompilation, which causes the backend to be offline for a few seconds).
4. Observe that the error will be logged within the console containing `Error: undefined`

**Versions**
```
System:
    OS: macOS 12.2.1
  Binaries:
    Node: 16.13.0 - ~/.nodenv/versions/16.13.0/bin/node
    Yarn: 1.22.17 - ~/.nodenv/versions/16.13.0/bin/yarn
    npm: 8.5.3 - ~/.nodenv/versions/16.13.0/bin/npm
  Browsers:
    Chrome: 97.0.4692.71
    Edge: 100.0.1185.29
    Safari: 15.3
  npmPackages:
    @apollo/client: 3.5.10 => 3.5.10 
    apollo-link-sentry: 3.0.2 => 3.0.2 
```

**Comment by mario-jerkovic** [comment]
Hi @davemo  👋, did you manage to "fix" this issue, or found a way around it?

**Comment by davemo** [comment]
> Hi @davemo 👋, did you manage to "fix" this issue, or found a way around it?

@mario-jerkovic no, I haven’t revisited this and we just kept using an older version. 

It may be that the bad ternary is still in the main/master branch.

**Comment by alessbell** [comment]
Hi @davemo and @mario-jerkovic 👋 We just shipped a fix for this issue in [v3.7.11](https://github.com/apollographql/apollo-client/releases/tag/v3.7.11) via https://github.com/apollographql/apollo-client/pull/10411. Thanks for the detailed report!
